File: app/src/main/java/com/example/test/di/AppComponent.java
package com.example.test.di;

import android.content.Context;
import com.example.test.presentation.HistoryBinActivity;
import com.example.test.presentation.MainActivity;

import java.util.Map;

import javax.inject.Provider;
import javax.inject.Singleton;

import androidx.lifecycle.ViewModel;
import dagger.BindsInstance;
import dagger.Component;

@Singleton
@Component(modules = {AppModule.class, DataBaseModule.class})
public interface AppComponent {

    Map<Class<? extends ViewModel>, Provider<ViewModel>> getMap();

    @Component.Factory
    interface Factory {
        AppComponent create(@BindsInstance Context context);
    }

    void inject(MainActivity mainActivity);

    void inject(HistoryBinActivity historyBinActivity);
}

File: app/src/main/java/com/example/test/di/AppComponent.kt
package com.example.test.di

import android.content.Context
import androidx.lifecycle.ViewModel
import com.example.test.presentation.HistoryBinActivity
import com.example.test.presentation.MainActivity
import dagger.BindsInstance
import dagger.Component
import javax.inject.Provider
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class, DataBaseModule::class])
interface AppComponent {

    fun getMap(): Map<Class<out ViewModel>, Provider<ViewModel>>

    @Component.Factory
    interface Factory {
        fun create(@BindsInstance context: Context):AppComponent
    }

    fun inject(mainActivity: MainActivity)

    fun inject(historyBinActivity: HistoryBinActivity)
}